src/bp-moderation/bp-moderation-filters.php

```diff
@@ -85,6 +83,72 @@ function bp_moderation_content_report() {
 add_action( 'wp_ajax_bp_moderation_content_report', 'bp_moderation_content_report' );
 add_action( 'wp_ajax_nopriv_bp_moderation_content_report', 'bp_moderation_content_report' );
 
+/**
+ * Function to handle frontend unblock user request.
+ *
+ * @since BuddyBoss 1.5.4
+ */
+function bp_moderation_unblock_user() {
+	$response = array(
+		'success' => false,
+		'message' => '',
+	);
+
+	$nonce = filter_input( INPUT_POST, 'nonce', FILTER_SANITIZE_STRING );
+	$type  = filter_input( INPUT_POST, 'type', FILTER_SANITIZE_STRING );
+	$id    = filter_input( INPUT_POST, 'id', FILTER_SANITIZE_NUMBER_INT );
+
+	if ( wp_verify_nonce( $nonce, 'bp-unblock-user' ) ) {
+
+		$unlock = bp_moderation_delete_reported_item( $id, $type );
+
+		if ( ! empty( $unlock ) ) {
+			$response['success'] = true;
+			$response['message'] = esc_html__( 'User unblocked successfully', 'buddyboss' );
+		}
+	}
+
+	if ( empty( $response['success'] ) && empty( $response['message'] ) ) {
+		$response['message'] = new WP_Error( 'bp_moderation_block_error', esc_html__( 'Sorry, Something happened wrong', 'buddyboss' ) );
+	}
+
+	echo wp_json_encode( $response );
+	exit();
+}
+
+add_action( 'wp_ajax_bp_moderation_unblock_user', 'bp_moderation_unblock_user' );
+add_action( 'wp_ajax_nopriv_bp_moderation_unblock_user', 'bp_moderation_unblock_user' );
+
+function bp_moderation_hide_request() {
```

Comment:
@janakkaneriya007 Funcation doing hide/unhide action both so funcation name and action should be like `bp_moderation_content_actions_request`